client/src/app/project/[projectid]/searchComponents/SearchResults.tsx

```diff
@@ -24,6 +24,57 @@ const SearchResults: FC<Props> = ({
   handleDrag,
   scheduleColors,
 }) => {
+  const [expandSort, setExpandSort] = useState(false);
+  const [expandFilter, setExpandFilter] = useState(false);
+  const [data, setData] = useState(noteData);
```

Comment:
I think there may be side effect issues with setting the initial state of the data here as the noteData? But i'm not 100% sure. The data flow goes as follows:

1. API is called to search for the data. It gets added to a state in the parent component called `allLocationData`. 
2. `allLocationData` gets broken down into 4 different page datas for where each of the modules use: `mapData`, `noteData`, `scheduleData`, `scheduleConfig`. For each search addition, these states get modified.
3. The resulting data gets passed down to each module for use directly.

You mentioned in discord that `mapData` persists after filtering out the information and its because the `mapData` would need to be updated as well.

I think it might be easier to move all of this code out into the parent component so we don't have to duplicate the data and stick with one source of truth originating from that parent component? What do you think?

_Sorry but I should have explained the code a bit earlier because it might be confusing and I didn't write any documentation ><. Let me know if you want to have a call so I can go through the architecture. We can also decide to refactor if you can think of a better way of writing this too!_

Reply:
Ah I didn't realize the data gets split up like that, I think it would be better for me to move the logic to the parent and use the allLocationData? I'll review the code more and I'll let you know if I need more of an explanation.

Reply:
Sorry - was my fault for not explaining how that worked!